cmd/multiFaSequenceSwap/multiFaSequenceSwap.go

```diff
@@ -0,0 +1,103 @@
+// input of this function includes the multiFa file name, the index of the background and foreground sequences from the multiFa, and a bed object with the relevant
+// start and stop indices of the swap regions
+// it will return a file containing a Fasta slice (which includes several sequences), with one more sequence that the sequences in the original
+// multiFa; this extra sequence is the background sequence with relevant foreground spots swapped as dictated by the bed file.
+func multiFaSubsequenceSwap(inFile string, swapRegionsFile string, backgroundName, foregroundName string, outFile string, outSeqName string) {
+	var currStart, currEnd, currPos int //these variables will be used to swap regions later in the code
+	// Load the original sequences from the multiFa file.
+	originalSeqs := fasta.Read(inFile)
+	swapRegions := bed.Read(swapRegionsFile)
+	background := getFaIndex(originalSeqs, backgroundName)
+	foreground := getFaIndex(originalSeqs, foregroundName)
+	// Ensure the specified background and foreground sequence indices are within the valid range.
+	if background < 0 || background >= len(originalSeqs) || foreground < 0 || foreground >= len(originalSeqs) {
+		log.Fatalf("Error: Invalid sequence index.\n")
+
+	}
+	answerSeq := fasta.Copy(originalSeqs[background]) //copy of the background sequence is made
+	for _, swapRegion := range swapRegions {          //loop over every single swap region in the bed
+		currStart = swapRegion.ChromStart
+		currEnd = swapRegion.ChromEnd
+
+		// Validate the swap region.
+		if currStart < 0 || currEnd > len(answerSeq.Seq) || currStart >= currEnd {
+			log.Fatalf("Error: Invalid swap region. \n")
+
+		}
+
+		// Swap the corresponding portion of answerSeq with the foreground sequence.
+		for currPos = currStart; currPos < currEnd; currPos++ {
+			answerSeq.Seq[currPos] = originalSeqs[foreground].Seq[currPos]
+		}
+	}
+	var finalSeqs []fasta.Fasta
+	answerSeq.Name = outSeqName
+	finalSeqs = append(originalSeqs, answerSeq)
+
+	fasta.Write(outFile, finalSeqs)
+
+}
+func getFaIndex(records []fasta.Fasta, name string) int {
+	for currIndex := 0; currIndex < len(records); currIndex++ {
+		if records[currIndex].Name == name {
+			return currIndex
+		}
+	}
+	log.Fatalf("Error: The requested sequence name: %s does not exist in the provided input. \n", name)
+	return -1
+
+}
+
+func usage() {
+	fmt.Print(
+		"multiFaSequenceSwap - Swaps regions of a sequence with regions from a different corresponding sequence. \n" +
+			"foregroundName is the name of the sequence that includes the regions in the bed elements.\n" +
+			"backgroundName is the name of the sequence that includes the regions outside the bed elements.\n" +
+			"Input of this function includes the multiFa file name,\n" +
+			"the index of the background and foreground sequences from the multiFa,\n" +
+			"a bed object with the relevant start and stop indices of the swap regions,\n" +
```

Comment:
I think it is a little unclear what the bed file represents.  I think most people will think it is for coordinates in the reference/first genome in the alignment, but I think this uses the bed coordinates as alignment positions (and ignores the name).  Maybe the code should be adjusted to have the bed represent reference coordinates, or maybe bed is not the best fit for the input format?

Reply:
Hi Craig, thanks for looking at this! I just pushed a commit that changes the program to handle the bed as reference positions, not as alignment positions